plugin-api/src/main/java/org/hyperledger/besu/plugin/data/PluginBlockSimulationResult.java

```diff
@@ -40,6 +44,29 @@ public PluginBlockSimulationResult(
     this.blockBody = blockBody;
     this.receipts = receipts;
     this.transactionSimulationResults = transactionSimulationResults;
+    this.trieLog = Optional.empty();
+  }
+
+  /**
+   * Constructs a new BlockSimulationResult instance with trielog.
+   *
+   * @param blockHeader the block header
+   * @param blockBody the block body
+   * @param receipts the list of transaction receipts
+   * @param transactionSimulationResults the list of transaction simulation results
+   * @param trieLog the optional trielog
+   */
+  public PluginBlockSimulationResult(
+      final BlockHeader blockHeader,
+      final BlockBody blockBody,
+      final List<? extends TransactionReceipt> receipts,
+      final List<TransactionSimulationResult> transactionSimulationResults,
+      final Optional<TrieLog> trieLog) {
+    this.blockHeader = blockHeader;
+    this.blockBody = blockBody;
+    this.receipts = receipts;
+    this.transactionSimulationResults = transactionSimulationResults;
+    this.trieLog = trieLog;
```

Comment:
AFAIK, Linea is the only downstream user of this service. Sounds good to me to return the actual trielog here. The consumer will ignore if there is no need